### DISCUSSION TITLE
Compaction bug in 0.12.2

### DISCUSSION THREAD

**Discussion by THUMarkLau** [apache/iotdb#3935] [Q&A]
Under some schema, the cross space compaction will report error when getting schema of timeseries. For example, if we have timeseries like:
```
root.FJXNY.350112.CJHCG50111AN01RAW00
root.FJXNY.350112.MUUNB00221PA03SUM00
root.FJXNY.350112.CJHCE10221DG01RAW00
root.FJXNY.350112.F11.MUUQE00121PW01RAW00
root.FJXNY.350112.F11.MUUNB00321PA04SUM00
root.FJXNY.350112.F11.MUUNB00121PA02SUM00
root.FJXNY.350112.F11.CJHCS50111SP01AVG06
```
The cross compaction will report an error like
![3dca51001586ab9550951c0a4108881](https://user-images.githubusercontent.com/37140360/132617173-ee0a72c9-275f-4d3d-8328-9966a91f284f.png)
The reason for this bug is that the program will get the device node root.FJXNY.350112, then visit all his children, and save his children as MeasurementNode. But a child root.FJXNY.350112.F11 of root.FJXNY.350112 is not a MeasurementNode, but a DeviceNode, so an error will be reported.

**Comment by jun0315** [comment]
Maybe you can take a look at this issue: https://issues.apache.org/jira/browse/IOTDB-1690 , the reason may be similar, PR: https://github.com/apache/iotdb/pull/3971

Welcome to exchange and discuss together.
